core/src/test/java/io/confluent/rest/SpireTest.java

```diff
@@ -0,0 +1,74 @@
+package io.confluent.rest;
+
+import io.spiffe.provider.SpiffeSslContextFactory;
+import io.spiffe.spiffeid.SpiffeId;
+import io.spiffe.spiffeid.TrustDomain;
+import io.spiffe.svid.x509svid.X509Svid;
+import io.spiffe.workloadapi.DefaultX509Source;
+import io.spiffe.workloadapi.X509Source;
+import io.spiffe.bundle.x509bundle.X509Bundle;
+
+import javax.net.ssl.*;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.security.KeyStore;
+import java.security.cert.X509Certificate;
+import java.util.Set;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class SpireTest {
+
+    @Test
+    public void testServerWithSpiffeMtlsClient() throws Exception {
+        String response = callServer("https://localhost:8080/hello?name=fff");
+
+        
+        assertTrue(response.contains("Hello, fff!"));
+    }
+
+    private String callServer(String serverUrl) throws Exception {
+        DefaultX509Source.X509SourceOptions x509SourceOptions = DefaultX509Source.X509SourceOptions
+                .builder()
+                .spiffeSocketPath("tcp://127.0.0.1:31523")
```

Comment:
Please ignore the details of this test file; once we agree , then i will make this properly.